packages/realm-react/src/useQuery.tsx

```diff
@@ -26,25 +26,49 @@ type RealmClassType<T = any> = { new (...args: any): T };
 type QueryCallback<T> = (collection: Realm.Results<T>) => Realm.Results<T>;
 type DependencyList = ReadonlyArray<unknown>;
 
+type QueryHookOptions<T> = {
+  type: string;
+  query?: QueryCallback<T>;
+  keyPaths?: string[];
+};
+
+type QueryHookClassBasedOptions<T> = {
+  type: RealmClassType<T>;
+  query?: QueryCallback<T>;
+  keyPaths?: string[];
+};
```

Comment:
Just realized that we should probably allow `keyPaths?: string | string[]` rather than just `string[]` since we do that in the realm package.

Reply:
I updated the API to allow a single string too.